src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -1657,6 +1671,118 @@ static int vmr_status_query(struct platform_device *pdev)
 	return vmr_control_op(pdev, XGQ_CMD_VMR_QUERY);
 }
 
+static void clk_scaling_cq_result_copy(struct xocl_xgq_vmr *xgq,
+                                       struct xocl_xgq_vmr_cmd *cmd)
+{
+	struct xgq_cmd_cq_default_payload *payload =
+		(struct xgq_cmd_cq_default_payload *)&cmd->xgq_cmd_cq_payload;
+	struct xgq_cmd_cq_clk_scaling_payload *cs_payload=
+		(struct xgq_cmd_cq_clk_scaling_payload *)&xgq->xgq_cq_payload;
+
+	mutex_lock(&xgq->xgq_lock);
```

Comment:
you don't have to copy the payload again, because it has been copied out automatically into the xgq_cmd_cq_payload.

you can directly use any of those by:
((struct xgq_cmd_cq_cls_payload *)&cmd->xgq_cmd_cq_payload)->has_clk_scaling;
etc.

Reply:
See examples in clock and log_page_payload.